src/index.tsx

```diff
@@ -125,8 +131,37 @@ async function activatePlugin(
 ): Promise<void> {
   const { commands } = app;
   const trans = translator.load('jupyterlab');
-  const fileBrowserTracker = browserFactory.tracker;
   const api = new SchedulerService({});
+
+  // Try calling an API to verify that the server extension is actually installed
+  let serverExtensionOk = false;
+  api
+    .getJobs({ max_items: 1 })
+    .then(response => {
+      serverExtensionOk = true;
+    })
+    .catch((responseError: ServerConnection.ResponseError) => {
```

Comment:
Let's prefer the `async`/`await` syntax as much as possible and minimize nesting. Suggested refactor:

```tsx
try {
  await api.getJobs({ max_items: 0 })
} catch (e: unknown) {
  // should work because ResponseError is a class
  if (e instanceof ServerConnection.ResponseError) {
    const httpStatus = e.response.status
    if (httpStatus === 404) {
      showDialog({ ... })
      return;
    }
  }
}
```

The early return also lets us drop the temporary `serverExtensionOk` boolean, since all error cases do not reach the bottom of this block anyways.

Reply:
I put this `try`/`catch` logic in, but the API error doesn't get caught (even though a 404 error comes back) and I see an error in the browser console, "Uncaught (in promise) Error: (HTML source)"

Reply:
Huh, I thought `try { await ... } catch(e) { ... }`  corresponded exactly with `Promise.reject`. Looks like wrapping an `await` call only catches synchronous exceptions, and not asynchronous ones (e.g. ones thrown after a network response).

Sucks that there doesn't appear to be a better syntax without nesting for this case. Oh well, it's fine. Thanks for trying.